android/app/src/main/java/app/organicmaps/sdk/sync/SyncPrefs.java

```diff
@@ -0,0 +1,407 @@
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.stream.Collectors;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class SyncPrefs
+{
+  private static final String TAG = SyncPrefs.class.getSimpleName();
+
+  private static final String PREF_NAME_ACCOUNTS = "SyncAccounts";
+
+  private static final String PREF_KEY_ACCOUNTS = "ac";
+  private static final String PREF_KEY_LAST_ACCOUNT_ID = "lastId";
+  private static final String PREF_KEY_SYNC_INTERVAL = "interval";
+  private static final String PREF_KEY_LAST_RUN = "lastRun";
+  private static final String PREF_KEY_NC_POLL = "ncPollParams";
+
+  private static final String PREF_KEY_PREFIX_ENABLED = "enabled-";
+  private static final String PREF_KEY_PREFIX_LAST_SYNCED = "lastSynced-";
+  private static final String PREF_KEY_PREFIX_ERROR_INFO = "syncError-";
+
+  private static final long DEFAULT_SYNC_INTERVAL_MS = 24 * 3600_000;
+  @Nullable
+  private static SyncPrefs instance;
+  @NonNull
+  private final SharedPreferences prefsAccounts;
+
+  private final CopyOnWriteArrayList<SyncAccount> mAccounts;
+  private final List<LastSyncCallback> mLastSyncCallbacks = new ArrayList<>(2);
+  private final List<ErrorInfoCallback> mErrorInfoCallbacks = new ArrayList<>(2);
+  private final List<AccountsChangedCallback> mAccountsChangedCallbacks = new ArrayList<>(2);
+  private final List<AccountToggledCallback> mAccountToggledCallbacks = new ArrayList<>(2);
+  private final List<FrequencyChangeCallback> mFrequencyChangeCallbacks = new ArrayList<>(2);
+
+  private SyncPrefs(Application context)
+  {
+    prefsAccounts = context.getSharedPreferences(PREF_NAME_ACCOUNTS, Context.MODE_PRIVATE);
+    Set<String> storedAccounts = prefsAccounts.getStringSet(PREF_KEY_ACCOUNTS, Collections.emptySet());
+    mAccounts = new CopyOnWriteArrayList<>(storedAccounts.stream()
+                                               .map(accountStr -> {
+                                                 try
+                                                 {
+                                                   return SyncAccount.fromJson(new JSONObject(accountStr));
+                                                 }
+                                                 catch (JSONException e)
+                                                 {
+                                                   Logger.e(TAG, "Invalid stored account JSON: " + accountStr, e);
+                                                   return null;
+                                                 }
+                                               })
+                                               .filter(Objects::nonNull)
+                                               .collect(Collectors.toList()));
+  }
+
+  public static SyncPrefs getInstance(Context context)
+  {
+    if (instance == null)
+      instance = new SyncPrefs((Application) context.getApplicationContext());
+    return instance;
```

Comment:
`ApplicationContext` is also a `Context`. You don't need `Application` here.
Remove `import android.app.Application;` from this file